src/core/rewrite.ml

```diff
@@ -287,14 +287,30 @@ let bind_match : term -> term -> tbinder =  fun p t ->
 
 (* NOTE in [bind_match] we lift while matching for efficiency. *)
 
-(** [rewrite ps po t] rewrites according to the equality proved by [t] in  the
-    current goal of [ps].  The term [t] should have a type corresponding to an
-    equality. Every occurrence of the first instance of the left-hand side  is
-    replaced by the right-hand side of the obtained proof. It also handles the
-    full set of SSReflect patterns. *)
-let rewrite
-    : Sig_state.t -> popt -> Proof.t -> rw_patt option -> term -> term =
-  fun ss pos ps p t ->
+(** [swap cfg a r l t] returns a term of type [P (eq a l r)] from a term [t]
+   of type [P (eq a r l)]. *)
+let swap : eq_config -> term -> term -> term -> term -> term =
+  fun cfg a r l t ->
+  (* NOTE The proofterm is “eqind a r l M (λx,eq a l x) (refl a l)”. *)
+  (* We build the predicate (“λx, eq a r x” in the above). *)
+  let pred =
+    let x = Bindlib.new_var mkfree "x" in
```

Comment:
Shouldn't we use a variable name that can't be input by the user, so that there cannot be name conflicts?